### PR TITLE
[threads] Allow resetting the name of a threadpool thread

### PR DESCRIPTION
**Pull request by ludovic-henry**
Fixes bug https://bugzilla.xamarin.com/show_bug.cgi?id=47221

### CODE REVIEW

**Comment by marek-safar** [comment]
Please add appropriate test

**Comment by ludovic-henry** [comment]
@marek-safar :white_check_mark:

File: mcs/class/corlib/Test/System.Threading/ThreadTest.cs [hunk 1]
```diff
@@ -908,6 +909,25 @@ public void CurrentThread_Domains ()
 		}
 #endif // MONO_FEATURE_MULTIPLE_APPDOMAINS
 
+		[Test]
+		public void SetNameInThreadPoolThread ()
+		{
+			for (int i = 0; i < 10; ++i) {
```
Comment (akoeplinger) [inline mcs/class/corlib/Test/System.Threading/ThreadTest.cs]: curious: why do we need to do this 10 times in the test, is there some sort of randomness that affects the bug?

Comment (ludovic-henry) [inline mcs/class/corlib/Test/System.Threading/ThreadTest.cs]: You want to run on multiple threading threads, and multiple times on the same one.

**Pull request merged**